app/scripts/controllers/HomepageCtrl.js

```diff
@@ -1,18 +1,48 @@
 (function() {
-    function HomepageCtrl(Room, Message) {
-        
+    function HomepageCtrl(Room, Message, BlocChatCookies) {
+
         this.rooms = Room;
-        
+
         this.addRoom = function(newRoomName) {
             Room.addRoom(newRoomName); //need to figure out how to clear newRoomName
         };
-        
+
         this.messages = Message;
-        
+
         this.activeRoomMessages = '';
+
+        this.activeRoomId = '';
+
+        this.cookies = BlocChatCookies;
+
+        this.setUsername = function(newUsername) {
```

Comment:
@jasonli12, since this function only does one thing (calls the `setUsername` function from the `BlocChatCookies` service), we could just call that function directly from the view instead of creating this extra function. I'll add another comment in the view to demonstrate what I mean.